core/lib/async-data-loader.ts

```diff
@@ -35,7 +35,8 @@ export const AsyncDataLoader = {
   },
   flush: function (actionContext: AsyncDataLoaderActionContext) {
     if (!actionContext.category) actionContext.category = DEFAULT_ACTION_CATEGORY
-    const actionsToExecute = this.queue.filter(ac => (!ac.category || !actionContext.category) || (ac.category === actionContext.category && (!ac.executedAt))).map(ac => {
+    const actionsWithDefaultCategory = this.queue.filter(ac => (!ac.category || !actionContext.category) || (ac.category === actionContext.category && (!ac.executedAt)));
+    const actionsToExecute = actionsWithDefaultCategory.map(ac => {
```

Comment:
Look at line 46. `.then(results => {
      return results
    })` part does nothing. Remove it